### DISCUSSION TITLE
MPGe stats

### DISCUSSION THREAD

**Discussion by 3rdStng** [teslamate-org/teslamate#2645] [Ideas]
I know we have the Wh/mi details, but is there an easy way to determine what our MPGe is based on this? I've been looking for a few days on how to calculate this and all I find is that 33.7 kWh = 1 US gallon of gas. Unless I am over thinking this, there doesn't appear to be an easy way to take the stats we have and determine our MPGe. Or like I said, I'm over thinking it. A quick little MPGe box in our Drives stats would be great. The reason for this is my father in-law drove my M3 the other day and was shocked that he started with 250 miles and ended with only 80 miles of charge. He only drove 101 miles, but he admits he has a very heavy foot too. I wanted to let him know what his MPGe was vs the 78% efficiency that was noted within TeslaMate.

**Comment by 3rdStng** [comment]
I think I was Googling this wrong in the past. Just found the answer, which is super simple to compute. And would be awesome to put into the graphs. 

In short, take 33,700 and divide that by your Wh/mi. One of my trips yesterday was 270 Wh/mi, as noted in my Drives dashboard. 
33700 / 270 = 124.8 MPGe.

I'm not a Grafana expert, but it seems like it should be easy to add a column in the Drives table to show this.
![image](https://user-images.githubusercontent.com/24888063/227665632-6f33828a-d3c7-45b8-8f30-fcd3d2d7ac7c.png)

**Reply by metaben** [reply]
In context though, MPGe is a distinctly American thing.
